pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -928,4 +945,14 @@ public String getSegmentName() {
   public int getMaxTimeForConsumingToOnlineSec() {
     return _maxTimeForConsumingToOnlineSec;
   }
+
+  private SegmentPartitionConfig createPartitionConfig() {
+    Map<String, ColumnPartitionConfig> map = new HashedMap();
+    for (String column : _column2PartitionersMap.keySet()) {
+      map.put(column, new ColumnPartitionConfig(
+          _column2PartitionersMap.get(column) + ColumnPartitionConfig.PARTITIONER_DELIMITER + _nPartitions,
+          "[" + _kafkaPartitionId + " " + _kafkaPartitionId + "]"));
```

Comment:
It would be good to have this logic separated somewhere else instead of passing Strings around. For example, it would make better sense design-wise to have something like `new ColumnPartitionConfig(partitioner, partitionCount, PartitionRange.build(kafkaPartitionId, kafkaPartitionId))` instead of expecting a specific string format.

Since there's already a billion comments on the PR, we can do it in a different PR.